### PR TITLE
Fix wrong assignment in computeMetricVariance for Frequency

### PR DESCRIPTION
**Pull request by riemanli**

### CODE REVIEW

**Comment by wfa-reviewable** [comment]
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464)

**Comment by riemanli** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #1464** <a href="https://app.graphite.dev/github/pr/world-federation-of-advertisers/cross-media-measurement/1464?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/world-federation-of-advertisers/cross-media-measurement/1464?utm_source=stack-comment).

**Review by jiayu-google — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @chenweiw and @riemanli)

___
*[`src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt` line 3354 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464#-NpzlpxwBjbxW0mfBv_T:-NpzlpxwBjbxW0mfBv_U:byu494e) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/fb6f13f32f702efadaf5aa671627eebb9a1fa360/src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt#L3354)):*
> ```Kotlin
>     val maximumFrequency = 5
>     val relativeFrequencyDistribution =
>       (1..maximumFrequency).associateWith { (maximumFrequency - it) / 10.0 }
> ```

where is 10.0 from?

File: src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt [hunk 1]
```diff
@@ -3330,6 +3330,111 @@ class VariancesTest {
...
+      ReachMeasurementVarianceParams(totalReach, reachMeasurementParams)
+    val reachMeasurementVariance =
+      VariancesImpl.computeMeasurementVariance(
+        DeterministicMethodology,
+        reachMeasurementVarianceParams,
+      )
+
+    val maximumFrequency = 5
+    val relativeFrequencyDistribution =
+      (1..maximumFrequency).associateWith { (maximumFrequency - it) / 10.0 }
```
Comment (chenweiw) [inline src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt:3354]: Line 3354: what does this line mean?

File: src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt [hunk 2]
```diff
@@ -3330,6 +3330,111 @@ class VariancesTest {
...
+    val expectedNKPlus =
+      listOf(
+          1978861168399.0,
+          2.5828737279307992e+23,
+          4.261741614272709e+23,
+          4.2617415905271664e+23,
+          2.582873680435757e+23,
+        )
+        .map { it * coefficient }
+
```
Comment (chenweiw) [inline src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt:3412]: I'm trying to understand the number of the four lists in line 3388--3409. Where do these expected values come from?

File: src/main/kotlin/org/wfanet/measurement/measurementconsumer/stats/Variances.kt [hunk 3]
```diff
@@ -627,7 +627,7 @@ object VariancesImpl : Variances {
     return FrequencyVariances(
       relativeVariances = frequencyVariances.relativeVariances.mapValues { coefficient * it.value },
       kPlusRelativeVariances =
-        frequencyVariances.relativeVariances.mapValues { coefficient * it.value },
+        frequencyVariances.kPlusRelativeVariances.mapValues { coefficient * it.value },
```
Comment (chenweiw) [inline src/main/kotlin/org/wfanet/measurement/measurementconsumer/stats/Variances.kt:630]: So this addressed the issue that we discussed yesterday -- the stds of rk and rk+ estimations are always equal. Am I right?

**Review by chenweiw — COMMENTED** [review]
Hi Rieman, I just made a couple of minor comments.

**Review by riemanli — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464)* status: 0 of 2 files reviewed, 4 unresolved discussions (waiting on @chenweiw and @jiayu-google)

___
*[`src/main/kotlin/org/wfanet/measurement/measurementconsumer/stats/Variances.kt` line 630 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464#gh-1480431974:-Nq-QICN-EneTeMwmIho:blph1wa) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/fb6f13f32f702efadaf5aa671627eebb9a1fa360/src/main/kotlin/org/wfanet/measurement/measurementconsumer/stats/Variances.kt#L630)):*
<details><summary><i>Previously, chenweiw wrote…</i></summary><blockquote>

So this addressed the issue that we discussed yesterday -- the stds of rk and rk+ estimations are always equal. Am I right?
</blockquote></details>

Yes. The fix here is using the correct field in `frequencyVariances`
___
*[`src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt` line 3354 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464#-NpzlpxwBjbxW0mfBv_T:-Nq-QlJVEusL92yMh4nv:b-swwfa) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/fb6f13f32f702efadaf5aa671627eebb9a1fa360/src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt#L3354)):*
<details><summary><i>Previously, jiayu-google wrote…</i></summary><blockquote>

where is 10.0 from?
</blockquote></details>

10.0 is the normalization term
___
*[`src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt` line 3354 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464#gh-1480423683:-Nq-QrBv9fThIYIixT7k:b-hbwqg6) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/fb6f13f32f702efadaf5aa671627eebb9a1fa360/src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt#L3354)):*
<details><summary><i>Previously, chenweiw wrote…</i></summary><blockquote>

Line 3354: what does this line mean?
</blockquote></details>

It creates a frequency distribution of [(5 - 1) / 10, (5 - 2) / 10, (5 - 3) / 10, (5 - 4) / 10, (5 - 5) / 10] = [0.4, 0.3, 0.2, 0.1, 0]
___
*[`src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt` line 3412 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464#gh-1480429757:-Nq-RLNt0H8Bx-vzc49E:b6e0477) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/fb6f13f32f702efadaf5aa671627eebb9a1fa360/src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt#L3412)):*
<details><summary><i>Previously, chenweiw wrote…</i></summary><blockquote>

I'm trying to understand the number of the four lists in line 3388--3409. Where do these expected values come from?
</blockquote></details>

These values are generated from the Colab notebook you and Jiayu wrote. Basically, I use the notebook as a ground truth generator to ensure the code in prod works the same as the code in the Colab notebook.

**Review by chenweiw — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464)* status: all files reviewed, 1 unresolved discussion (waiting on @jiayu-google)

**Review by riemanli — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464)* status: all files reviewed, 1 unresolved discussion (waiting on @jiayu-google and @Marco-Premier)

___
*[`src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt` line 3354 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464#-NpzlpxwBjbxW0mfBv_T:-Nq-Uha46XNY3DaSvov3:b9p6sry) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/fb6f13f32f702efadaf5aa671627eebb9a1fa360/src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt#L3354)):*
<details><summary><i>Previously, riemanli (Rieman) wrote…</i></summary><blockquote>

10.0 is the normalization term
</blockquote></details>

It creates a frequency distribution of \[(5 - 1) / 10, (5 - 2) / 10, (5 - 3) / 10, (5 - 4) / 10, (5 - 5) / 10\] = \[0.4, 0.3, 0.2, 0.1, 0\]

**Review by Marco-Premier — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464)* status: all files reviewed, 1 unresolved discussion (waiting on @jiayu-google and @stevenwarejones)

**Review by stevenwarejones — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464)* status: all files reviewed, 1 unresolved discussion (waiting on @jiayu-google)

**Review by riemanli — COMMENTED** [review]
> Dismissed @jiayu-google from a discussion.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1464)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @jiayu-google)

**Pull request merged**